src/components/ClusterManager.js

```diff
@@ -550,22 +619,33 @@ export default ajaxCaller(class ClusterManager extends PureComponent {
           icon('cog', { size: 22, className: 'is-solid', style: { color: isDisabled ? colors.gray[2] : colors.green[0] } })
         ])
       ]),
-      deleting && h(Modal, {
+      deleteModalOpen && h(Modal, {
         title: 'Delete notebook runtime?',
-        onDismiss: () => this.setState({ deleting: false }),
+        onDismiss: () => this.setState({ deleteModalOpen: false }),
         okButton: () => {
-          this.setState({ deleting: false })
+          this.setState({ deleteModalOpen: false })
           this.destroyActiveCluster()
         }
       }, ['Deleting the cluster will stop all running notebooks and associated costs. You can recreate it later, which will take several minutes.']),
-      open && h(NewClusterModal, {
+      createModalOpen && h(NewClusterModal, {
         namespace, currentCluster,
-        onCancel: () => this.setState({ open: false }),
+        onCancel: () => this.setState({ createModalOpen: false }),
         onSuccess: promise => {
-          this.setState({ open: false })
-          this.executeAndRefresh(promise)
+          this.setState({ createModalOpen: false })
+          const doCreate = async () => {
+            if (currentCluster.status === 'Error') {
+              const { googleProject, clusterName } = this.getCurrentCluster()
+              await Jupyter.cluster(googleProject, clusterName).delete()
+            }
+            return promise
+          }
+          this.executeAndRefresh(doCreate())
```

Comment:
This isn't the only place that uses `NewClusterModal`, and presumably we want this behavior to be the same across all usages. So instead of doing this here, we could put this logic into `createCluster` inside the modal component. (It can use `Promise.all` to aggregate the creation and deletion promises into one)